share/functions/fish_delta.fish

```diff
@@ -0,0 +1,100 @@
+function fish_delta
+    argparse h/help f/no-functions c/no-completions d/no-diff n/new -- $argv
+
+    if set -q _flag_help
+        __fish_print_help fish_delta
+        return 0
+    end
+
+    type -q diff
+    or set -l _flag_d --no-diff
+
+    # TODO: Do we want to keep the vendor dirs in here?
+    set -l default_function_path $__fish_data_dir/functions
+    set -l default_complete_path $__fish_data_dir/completions
+
+    set -l user_function_path
+    set -l user_complete_path
+    
+    for dir in $fish_function_path
+        contains -- $dir $default_function_path
+        or set -a user_function_path $dir
+    end
+    for dir in $fish_complete_path
+        # We don't care about generated completions.
+        # They shouldn't be compared at all.
+        contains -- $dir $default_complete_path $__fish_user_data_dir/generated_completions
+        or set -a user_complete_path $dir
+    end
+
+    set -l vars
+    not set -ql _flag_f[1]
+    and set -a vars user_function_path default_function_path
+    not set -ql _flag_c[1]
+    and set -a vars user_complete_path default_complete_path
+
+    if not set -q vars[1]
+        echo No directories to check! >&2
+        return 2
+    end
+    
+    set -l have_diff 0
+
+    if isatty stdout
+        set -f colors "$(set_color normal)" "$(set_color brblue)" "$(set_color bryellow)" "$(set_color green)"
+        set -f diff_color --color=always
+        set -f pager less
+        set -q PAGER
+        and echo $PAGER | read -at pager
+        set -l lessopts isRF
+        if test (less --version | string match -r 'less (\d+)')[2] -lt 530 2>/dev/null
+            set lessopts "$lessopts"X
+        end
+
+        not set -qx LESS
+        and set -xf LESS $lessopts
+    else
+        set -f colors "" "" "" ""
+        set -f diff_color
+        set -f pager cat # cannot use a builtin here
+    end
+
+    while set -q vars[1]
+        set -l user_var $$vars[1]
+        set -l default_var $$vars[2]
+        set -e vars[..2]
+        set -l files (path filter -rf -- $user_var/$argv.fish)
+        set -q argv[1]
+        or set files (path filter -rf -- $user_var/*)
+        set -q files[1]
+        and set have_diff 1
+
+        for file in $files
+            set -l bn (path basename -- $file)
+            set -l def (path filter -rf -- $default_var/$bn)[1]
+            or begin
+                set -ql _flag_n
+                and printf (_ "%sNew%s: %s\n") $colors[2] $colors[1] $file
+                continue
+            end
+
+            if type -q diff
+                # We execute diff twice - once to figure out if it's changed,
+                # so we can get nicer output.
+                #
+                if not diff -q -- $file $def >/dev/null 2>&1
```

Comment:
Why is it faster *in theory*?

We give `diff` enough information here that it knows we just want to know if there's a difference. That means it could quit early or even use whatever faster path `cmp` uses.

Regardless - this isn't performance-sensitive (it's quite fast in practice *and* it's okay if this takes a while), and I would rather depend on one command being present and about what we expect than two.

In fact I probably need to see if `--color=always` needs to be checked.